envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactory.kt

```diff
@@ -126,10 +131,28 @@ class RBACFilterFactory(
         return Permission.newBuilder().setHeader(methodMatch).build()
     }
 
-    private fun mapClientToPrincipal(client: String): Principal {
+    private fun mapClientToPrincipals(client: String, snapshot: GlobalSnapshot): List<Principal> {
+        val principals = if (client in incomingPermissionsProperties.sourceIpAuthentication.enabledForServices) {
+            snapshot.endpoints.resources()
+                    .filterKeys { client == it }.values
+                    .flatMap { clusterLoadAssignment ->
+                        clusterLoadAssignment.endpointsList.flatMap { lbEndpoints ->
```

Comment:
After my change if a client is not in global snapshot (meaning there are no endpoints of it so no one can reach it) it will not get a principal / policy so access will be denied.